lib/item.rb

```diff
@@ -0,0 +1,66 @@
+require 'httparty'
+require "pry"
+
+class Item
+  attr_reader :nypl_source, :id
+  attr_accessor :item_type_code, :location_code
+
+  def initialize(nypl_source, id)
+    @nypl_source = nypl_source
+    @id = id
+  end
+
+  def is_research?
+    get_platform_api_data
+
+    result = is_partner? || item_type_is_research? || location_is_only_research?
+
+    $logger.debug "Evaluating is-research for #{nypl_source} #{id}: #{result}", $log_data
+
+    return result
+  end
+
+  private
+    def is_partner?
+      result = nypl_source == "recap-cul" || nypl_source == "recap-pul"
+      $log_data = {is_partner?: result}
+      return result
+    end
+
+    def item_type_is_research?
+      item_collection_type = $nypl_core.by_catalog_item_type[item_type_code]
+      if item_collection_type.nil?
+        $logger.debug "Unknown item_type #{item_type_code}"
+        return false
+      end
+      result = item_collection_type["collectionType"].include?("Research")
+      $log_data[:item_type_is_research?] = result
+      return result
+    end
+
+    def location_is_only_research?
+      sierra_location = $nypl_core.by_sierra_location[location_code]
+      if sierra_location.nil?
+        $logger.debug "Unknown location_code #{location_code}"
+        return false
+      end
+      collection_types = sierra_location["collectionTypes"]
+      result = collection_types == ["Research"]
+      $log_data[:location_is_only_research] = result
+      return result
+    end
+
+    def get_platform_api_data
+      begin
+        response = $platform_api.get("items/" + @nypl_source + "/" + @id)
+        data = response["data"]
+
+        return if is_partner?
+
+        self.item_type_code = data["fixedFields"]["61"]["value"]
+        self.location_code = data["location"]["code"]
+      rescue => e
+        raise ParameterError.new(response["message"])
+      end
```

Comment:
The errors that can be thrown in the above block include:
 1. Network timeout or api credentials issue connecting to platform api (which should be a 500)
 1. Item record not found (which should be a 404)
 1. Item record found, but lacks item type/ location (which should be a 500)

If any of the above errors are encountered, I believe the above currently returns the same 400 level error.

I'd restructure it to ensure that low-level network timeouts on the platform api are handled as 500, lookup failures are handled as 404s (detectable by checking for missing "data" property in `response`), and property access issues (missing data) are handled as 500s.

I like the idea discussed elsewhere of adding a second customer error in `lib/errors` called something like `DataError` to throw mindfully when any necessary data is missing (e.g. record exists but has no "fixedFields" or no "fixedFields" "61" value)